### DISCUSSION TITLE
Maximizing certain windows on startup but not all windows

### DISCUSSION THREAD

**Discussion by alyzmal** [wezterm/wezterm#3702] [Q&A] [answered]
I've been trying to figure out how to set up the Lua so that upon clicking the desktop link to my Neovim setup (a setup whose command input is "wezterm start -- nvim"), it sets that window to fullscreen. While I've been able to do this, it comes with the unintended consequence of also making all other initial windows fullscreen, which is not what I want. Is this a case of me not understanding the commands? And if so, is there a way to set up the GUI in the way I've just described?

**Comment by wez** [comment]
You could inspect the `cmd` in the `gui-startup` event to see if it is nvim before maximizing
https://wezfurlong.org/wezterm/config/lua/gui-events/gui-startup.html

**Reply by alyzmal** [reply]
I have been trying on and off for several weeks to find the right combination of syntax to achieve this and I have had no luck. I am unsure of exactly which part of the spawn command I am supposed to inspect and in which order I must do this to ensure that only the requested windows become fullscreen. Any help would be very much appreciated.

**Reply by nekowinston** [reply, marked as answer]
This should work:
```lua
wezterm.on("gui-startup", function(cmd)
  if cmd.args[1] == "nvim" then
    local tab, pane, window = wezterm.mux.spawn_window(cmd)
    window:gui_window():maximize()
  end
end)
```

**Reply by alyzmal** [reply]
That is exactly what I was looking for! I couldn't figure out how to get the command to register the argument, but that fixed everything! Thank you so so much!

Edit: I will note that I had to add an extra check to make sure that the value of cmd.args was not nil to avoid a recurrent runtime error, but this was the base that I needed to get that far.
